common/lib/xmodule/xmodule/js/spec/video/video_bumper_spec.js

```diff
@@ -0,0 +1,109 @@
+(function (WAIT_TIMEOUT) {
+    'use strict';
+    describe('VideoBumper', function () {
+        var state, oldOTBD;
+
+        beforeEach(function () {
+            oldOTBD = window.onTouchBasedDevice;
+            window.onTouchBasedDevice = jasmine
+                .createSpy('onTouchBasedDevice').andReturn(null);
+            state = jasmine.initializePlayer('video_with_bumper.html');
+            $('.poster .btn-play').click();
+            jasmine.Clock.useMock();
+        });
+
+        afterEach(function () {
+            $('source').remove();
+            state.storage.clear();
+            if (state.bumperState && state.bumperState.videoPlayer) {
+                state.bumperState.videoPlayer.destroy();
+            }
+            if (state.videoPlayer) {
+                state.videoPlayer.destroy();
+            }
+            window.onTouchBasedDevice = oldOTBD;
+        });
+
+        it('can render the bumper video', function () {
+            expect($('.is-bumper')).toExist();
+        });
+
+        it('can show the main video on error', function () {
+            state.el.trigger('error');
+            jasmine.Clock.tick(20);
+            expect($('.is-bumper')).not.toExist();
+            waitsFor(function () {
+                return state.el.hasClass('is-playing');
+            }, 'Player is not playing.', WAIT_TIMEOUT);
+        });
+
+        it('can show the main video once bumper ends', function () {
+            state.el.trigger('ended');
+            jasmine.Clock.tick(20);
+            expect($('.is-bumper')).not.toExist();
+            waitsFor(function () {
+                return state.el.hasClass('is-playing');
+            }, 'Player is not playing.', WAIT_TIMEOUT);
+        });
+
+        it('can show the main video on skip', function () {
+            state.bumperState.videoBumper.skip();
+            jasmine.Clock.tick(20);
+            expect($('.is-bumper')).not.toExist();
+            waitsFor(function () {
+                return state.el.hasClass('is-playing');
+            }, 'Player is not playing.', WAIT_TIMEOUT);
+        });
+
+        it('can stop the bumper video playing if it is too long', function () {
+            state.el.trigger('timeupdate', [state.bumperState.videoBumper.maxBumperDuration + 1]);
+            jasmine.Clock.tick(20);
+            expect($('.is-bumper')).not.toExist();
+            waitsFor(function () {
+                return state.el.hasClass('is-playing');
+            }, 'Player is not playing.', WAIT_TIMEOUT);
```

Comment:
I think these 4 lines are repeated in the four tests; would be nice to extract them into a function.